### PR TITLE
Authentication in front end is fully functional.

### PR DESCRIPTION
**Pull request by wAndrewx**
Authentication in front end is fully functional.

Users can now:

- Log in
- Log out
- Register a new account
These changes can be seen in ui changes

### CODE REVIEW

File: components/Navbar.js [hunk 1]
```diff
@@ -5,28 +5,48 @@ import {
...
+import { logoutUser } from './authentication/authUtils';
 export const Navbar = () => {
   const { toggleColorMode } = useColorMode();
   const bg = useColorModeValue('#db524d', '#33332d');
   const text = useColorModeValue('dark', 'light');
   const SwitchIcon = useColorModeValue(MoonIcon, SunIcon);
-
   const { isOpen, onOpen, onClose } = useDisclosure();
-  // const [isMobile] = useMediaQuery('(min-width:800px)');
-  // console.log(isMobile);
```
Comment (khoaHyh) [inline components/Navbar.js:24]: remove console.log

File: components/Navbar.js [hunk 2]
```diff
@@ -5,28 +5,48 @@ import {
...
+  const [getLoginBool, setLoginBool] = useState(false);
+
+  useEffect(() => {
+    // onload set login to false
+    const landingSession =
+      JSON.parse(window.localStorage.getItem('isLoggedIn')) === null
+        ? false
+        : JSON.parse(window.localStorage.getItem('isLoggedIn'));
+    // console.log(landingSession);
```
Comment (khoaHyh) [inline components/Navbar.js:31]: and here

File: components/authentication/authUtils.js [hunk 3]
```diff
@@ -116,15 +126,17 @@ const userValidation = async (password = '', userName = '') => {
...
-  } else {
-    if (alphanumRegex.test(userName)) return true;
-    else return false;
-  }
+const registerUser = async (newUserSchema) => {
+  const res = await axios.post(
+    `${process.env.NEXT_PUBLIC_API_URL}${'/register'}`,
+    newUserSchema
+  );
+  return res;
```
Comment (khoaHyh) [inline components/authentication/authUtils.js:139]: Are we handling errors elsewhere? If not add a try catch block at the level where errors for `login` and `logout` are being handled.

File: components/authentication/login/LoginPanel.js [hunk 4]
```diff
@@ -4,29 +4,35 @@ import {
...
 
 export const LoginPanel = () => {
-  const [message, setMessage] = useState('');
+  const [message, setMessage] = useState([]);
   const [isSuccess, setSuccess] = useState(false);
   const [userName, setUserName] = useState();
   const [password, setPassword] = useState();
 
-  // console.log(message);
-  // console.log(isSuccess);
```
Comment (khoaHyh) [inline components/authentication/login/LoginPanel.js:18]: remove console logs here

File: components/authentication/login/LoginPanel.js [hunk 5]
```diff
@@ -4,29 +4,35 @@ import {
...
-
   const handleLogin = async e => {
     e.preventDefault();
     try {
+      // setMessage();
```
Comment (khoaHyh) [inline components/authentication/login/LoginPanel.js:21]: remove this

File: components/authentication/login/LoginPanel.js [hunk 6]
```diff
@@ -4,29 +4,35 @@ import {
...
-        `${process.env.NEXT_PUBLIC_API_URL}${'/login'}`
+        `${process.env.NEXT_PUBLIC_API_URL}${'/login'}`,
+        { username: userName, password: password }
       );
-      console.log(res);
+      setSuccess(res.data.success);
+      setMessage([{ message: res.data.message }]);
+      // console.log(res.data);
+      window.localStorage.setItem('isLoggedIn', true);
+      // }
```
Comment (khoaHyh) [inline components/authentication/login/LoginPanel.js:30]: the console log above and this commented out curly brace

File: components/authentication/login/RegisterPanel.js [hunk 7]
```diff
@@ -4,29 +4,25 @@ import {
...
 import axios from 'axios';
-import { userValidation } from '../PasswordValidation';
+import { userValidation, registerUser } from '../authUtils';
 import { StatusAlert } from './StatusAlert';
 
 export const RegisterPanel = () => {
   const [userName, setUserName] = useState();
   const [email, setEmail] = useState();
   const [password, setPassword] = useState();
-  // const [status, setStatus] = useState('');
```
Comment (khoaHyh) [inline components/authentication/login/RegisterPanel.js:18]: remove this

File: components/authentication/login/StatusAlert.js [hunk 8]
```diff
@@ -5,24 +5,34 @@ import {
...
-import { useRef } from 'react';
+import { useEffect, useRef, useState } from 'react';
 
+// Problem:
+// Component loads everytime a user types anything
+// Goal: Only updates on submit...
+// How: componentdidmount lifecycle...
 export const StatusAlert = ({ message, status }) => {
-  console.log('passthru:', message);
-  // status={status}
```
Comment (khoaHyh) [inline components/authentication/login/StatusAlert.js:12]: remove this

**Review by khoaHyh — CHANGES_REQUESTED** [review]
Added comments below changes that need to be made.

File: components/authentication/authUtils.js (hunk 3 above)
Comment (wAndrewx) [reply components/authentication/authUtils.js:139]: yes they are being handled when its being called

File: components/Navbar.js [hunk 9]
```diff
@@ -42,9 +51,7 @@ export const Navbar = () => {
       const res = await logoutUser();
       window.localStorage.setItem('isLoggedIn', false);
       setLoginBool(window.localStorage.getItem('isLoggedIn'));
-    } catch (error) {
-      console.log('error');
-    }
+    } catch (error) {}
```
Comment (khoaHyh) [inline components/Navbar.js:54]: Need to handle error here in side curly braces.

**Review by khoaHyh — CHANGES_REQUESTED** [review]
still need to remove some comments and fix stuff in comments

**Comment by wAndrewx** [comment]
Ok i believed i pushed in the commits to this pr. should be finsihed

File: components/Navbar.js (hunk 1 above)
Comment (wAndrewx) [reply components/Navbar.js:24]: done

File: components/Navbar.js (hunk 2 above)
Comment (wAndrewx) [reply components/Navbar.js]: done

**Pull request merged**